doc/openhim.md

```diff
@@ -0,0 +1,102 @@
+## Integrating with OpenHIM (Middleware component)
```

Comment:
nit: Let's have the top level title as `#` (instead of `##`) and first level sections as `##` (instead of `###`); here and in all other new docs.

```suggestion
# Integrating with OpenHIM (Middleware component)
```